yb-voyager/src/tgtdb/oracle.go

```diff
@@ -0,0 +1,520 @@
+			return rowsAffected, fmt.Errorf("run sqlldr: %w", err)
+		}
+	}
+
+	err = tdb.recordEntryInDB(tx, batch, rowsAffected)
+	if err != nil {
+		err = fmt.Errorf("record entry in DB for batch %q: %w", batch.GetFilePath(), err)
+	}
+
+	return rowsAffected, err
+}
+
+func (tdb *TargetOracleDB) runSqlLdr(oracleConnectionString string, sqlldrArgs string, password string) (outbufStr string, errbufStr string, err error) {
+	var outbuf bytes.Buffer
+	var errbuf bytes.Buffer
+	cmd := exec.Command("sqlldr", sqlldrArgs)
+	stdinPipe, err := cmd.StdinPipe()
+	if err != nil {
+		return "", "", fmt.Errorf("get stdin pipe for sqlldr: %w", err)
+	}
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+
+	err = cmd.Start()
+	if err != nil {
+		return "", "", fmt.Errorf("start sqlldr: %w", err)
+	}
+	_, err = stdinPipe.Write([]byte(fmt.Sprintf("%s\n", password)))
+	if err != nil {
+		return "", "", fmt.Errorf("write password to sqlldr: %w", err)
+	}
+	err = stdinPipe.Close()
+	if err != nil {
+		return "", "", fmt.Errorf("close stdin pipe for sqlldr: %w", err)
+	}
+	err = cmd.Wait()
+
+	return outbuf.String(), errbuf.String(), err
+}
+
+func (tdb *TargetOracleDB) recordEntryInDB(tx *sql.Tx, batch Batch, rowsAffected int64) error {
+	cmd := batch.GetQueryToRecordEntryInDB(tdb.tconf.TargetDBType, rowsAffected)
+	_, err := tx.ExecContext(context.Background(), cmd)
+	if err != nil {
+		return fmt.Errorf("insert into %s: %w", BATCH_METADATA_TABLE_NAME, err)
+	}
+	return nil
+}
+
+func getRowsAffected(outbuf string) (int64, error) {
+	regex := regexp.MustCompile(`Load completed - logical record count (\d+).`)
+	matches := regex.FindStringSubmatch(outbuf)
+	if len(matches) < 2 {
+		return 0, fmt.Errorf("no rows affected found in the sqlldr output")
+	}
+	return strconv.ParseInt(matches[1], 10, 64)
+}
+
+func (tdb *TargetOracleDB) isBatchAlreadyImported(tx *sql.Tx, batch Batch) (bool, int64, error) {
+	var rowsImported int64
+	query := batch.GetQueryIsBatchAlreadyImported(tdb.tconf.TargetDBType)
+	err := tx.QueryRowContext(context.Background(), query).Scan(&rowsImported)
+	if err == nil {
+		log.Infof("%v rows from %q are already imported", rowsImported, batch.GetFilePath())
+		return true, rowsImported, nil
+	}
+	if err == sql.ErrNoRows {
+		log.Infof("%q is not imported yet", batch.GetFilePath())
+		return false, 0, nil
+	}
+	return false, 0, fmt.Errorf("check if %s is already imported: %w", batch.GetFilePath(), err)
+}
+
+func (tdb *TargetOracleDB) setTargetSchema(conn *sql.Conn) {
+	checkSchemaExistsQuery := fmt.Sprintf(
+		"SELECT 1 FROM ALL_USERS WHERE USERNAME = '%s'",
+		strings.ToUpper(tdb.tconf.Schema))
+	var cntSchemaName int
+
+	if err := conn.QueryRowContext(context.Background(), checkSchemaExistsQuery).Scan(&cntSchemaName); err != nil {
+		utils.ErrExit("run query %q on target %q to check schema exists: %s", checkSchemaExistsQuery, tdb.tconf.Host, err)
+	} else if cntSchemaName == 0 {
+		utils.ErrExit("schema '%s' does not exist in target", tdb.tconf.Schema)
+	}
```

Comment:
This should be checked only once.

Consider moving the check for existance of target schema in the `TargetDB.Init()`.

Consider making similar change for YB as well.